### PR TITLE
refactor: BFF contract updates (round 2)

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** A bunch of things needed to change on the contract, see details below.

JIRA:  [AU-1481](https://2u-internal.atlassian.net/browse/AU-1481)

**What changed?**

- Remove `rubric` from `progress`. (3e3c045595c5f46e36539e8b69e08c02873384c8)
- Remove `hasReceivedGrade` from `submission`. (e7f0c35540f1e6a9abfd8731f54bb5064379d3b0)
- Move `submission` status from `submission` into `stepInfo`. (32b0f141870d5e64f91deefdd2ca59e85b324fab)
    - Add `isClosed` and `closedReason` to `submission`.
- Rename `submission` to `response` (063d2a8f7c19d9cccbce8839c17d43061ce1d3bb)
- Combine `get_learner_block_submission`/`assessments_data` into single `get_learner_data` call (4723d8ef3af32157666b11f4c7f8c3aa15ad39d5)
- Move `teamInfo` from `response` info`stepInfo`. (b8f2896133eae7d0dcf23f9a1e212bce9c72153b)
- Move team file uploads to `response`. (47847abbaef049194afa8591ecf7624bc79acae4)
- Move `response` up a level (d517e538e227c842cc109b312ef8859db9d63ea6)
- Update all `startTime` / `endTime` to `startDatetime` and `endDatetime`. (f84ac3129035598b0457ff1a3951e7a01bd6c9c9)
- Other various cleanups & refactors.
    - Always show peer and submission step info once we've reached those steps.

**Testing Instructions**

- Unit tests
- Use Postman or other API tool to exercise `{{protocol}}://{{lms_url}}/courses/{{course_id}}/xblock/{{block_id}}/handler/get_learner_data/{{jump_step}}`

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/xblock/apis/workflow_api.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def status(self):
         return self.workflow.get("status")
 
     @property
-    def has_recieved_grade(self):
```
Comment (jansenk) [inline openassessment/xblock/apis/workflow_api.py:86]: lol

File: openassessment/xblock/ui_mixins/mfe/mixin.py [hunk 2]
```diff
@@ -48,25 +48,66 @@ def get_block_info(self, data, suffix=""):  # pylint: disable=unused-argument
...
+        Get data for the user / step of the ORA
+        - If no step provided, go to current workflow step
+            - If not submitted, load draft submission
+            - If submitted, load fir
+        - If user provides jumpable step
+            - Check if we can get to that step
+            - If we can, go to that step
+            - Otherwise, raise an error
+        """
+        workflow_step = self.workflow_data.status or "submission"
```
Comment (jansenk) [inline openassessment/xblock/ui_mixins/mfe/mixin.py:62]: what if the workflow step is 'done' or 'waiting'?

File: openassessment/xblock/ui_mixins/mfe/page_context_serializer.py [hunk 3]
```diff
@@ -48,6 +47,7 @@ class ReceivedGradesSerializer(Serializer):
     self = AssessmentScoreSerializer(source="grades.self_score")
     peer = AssessmentScoreSerializer(source="grades.peer_score")
     staff = AssessmentScoreSerializer(source="grades.staff_score")
+    # teams = AssessmentGradeSerializer(source="grades.teams_score")
```
Comment (jansenk) [inline openassessment/xblock/ui_mixins/mfe/page_context_serializer.py:50]: delete?

File: openassessment/xblock/ui_mixins/mfe/page_context_serializer.py [hunk 4]
```diff
@@ -237,58 +299,46 @@ class PageDataSerializer(Serializer):
...
-    def get_submission(self, instance):
+    def get_response(self, instance):
         """
         we get the user's draft / complete submission.
         """
         # pylint: disable=broad-exception-raised
+
         # Submission Views
         if self.context.get("view") == "submission":
             learner_page_data_submission_data = instance.get_learner_submission_data()
```
Comment (jansenk) [inline openassessment/xblock/ui_mixins/mfe/page_context_serializer.py:324]: since we've changed the name of the serializer, we can also change the name of this variable

File: openassessment/xblock/ui_mixins/mfe/page_context_serializer.py [hunk 5]
```diff
@@ -237,58 +299,46 @@ class PageDataSerializer(Serializer):
...
-            active_step = instance.workflow_data.status or "submission"
-            self.context.update({"step": active_step})
+        if "step" not in self.context:
+            raise ValidationError("Missing required context: step")
+        if "view" not in self.context:
+            raise ValidationError("Missing required context: view")
 
         return super().to_representation(instance)
 
-    def _can_jump_to_step(self, workflow_step, workflow_data, step_name):
```
Comment (jansenk) [inline openassessment/xblock/ui_mixins/mfe/page_context_serializer.py:256]: This is good, we should pull as much of this logic out of the serializer as we can

**Review by jansenk — APPROVED** [review]

**Comment by nsprenkle** [comment]
Holding, as I realized I also need to do assessment responses

**Review by mattcarter — APPROVED** [review]
approved
